File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.go
import (
	"sort"
)

func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	if len(nums1) > 0 && len(nums2) > 0 {
		if nums1[0] > nums2[0] {
			t := nums1
			nums1 = nums2
			nums2 = t
		}
	}

	nums := append(nums1, nums2...)
	if len(nums) == 0 {
		return float64(0)
	}
	sort.Ints(nums)
	half := len(nums) / 2

	if len(nums)%2 > 0 {
		return float64(nums[half])
	}
	return float64(nums[half-1]+nums[half]) / 2.0
}

File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        nums = nums1 + nums2
        
        if not nums:
            return float(0)
        
        nums.sort()
        mid = len(nums) // 2
        
        if len(nums) % 2 == 0:       
            return  (nums[mid-1] + nums[mid]) / 2
        else:
            return float(nums[mid])
            
        
        
    